motionplan/motionPlanner.go

```diff
@@ -577,24 +567,20 @@ func CheckPlan(
 			if currentTravelDistanceMM > lookAheadDistanceMM {
 				return nil
 			}
-			// If we are working with a PTG plan the returned value for poseInPath will only
-			// tell us how far along the arc we have traveled. Since this is only the relative position,
-			// i.e. relative to where the robot started executing the arc,
-			// we must compose poseInPath with currentPose to get the absolute position.
-			// In both cases we ultimately compose with errorState.
+
+			// Checks for collision along the interpolated route and returns a the first interpolated pose where a collision is detected.
+			interpolatedState := &ik.State{Frame: sf}
 			if relative {
-				rectifyBy := spatialmath.Compose(currentPose, errorState)
-				poseInPath = spatialmath.Compose(rectifyBy, poseInPath)
+				interpolatedState.Position = spatialmath.Compose(segment.StartPosition, poseInPath)
 			} else {
-				poseInPath = spatialmath.Compose(poseInPath, errorState)
+				interpolatedState.Configuration = interpConfig
 			}
```

Comment:
They both pass the existing tests and furthermore I have inspected the interpolatedState in detail for the test cases.  

The reason this change was needed is because now the Path is stored in terms of absolute coordinates instead of being "rectified".  For the non-relative case it is not as useful to store the Pose info as it is less descriptive than the configuration info